src/garage/np/_functions.py

```diff
@@ -368,3 +370,34 @@ def slice_nested_dict(dict_or_array, start, stop):
         # It *should* be a numpy array (unless someone ignored the type
         # signature).
         return dict_or_array[start:stop]
+
+
+def pad_batch_array(array, lengths, max_length=None):
+    r"""Convert a packed into a padded array with one more dimension.
+
+    Args:
+        array (np.ndarray): Array of length :math:`(N \bullet [T], X^*)`
+        lengths (list[int]): List of length :math:`N` containing the length
+            of each episode in the batch array.
+        max_length (int): Defaults to max(lengths) if not provided.
+
+    Returns:
+        np.ndarray: Of shape :math:`(N, max_length, X^*)`
+
+    """
+    assert array.shape[0] == sum(lengths)
+    if max_length is not None and max_length > max(lengths):
+        # This is probably not a good idea, but RL2 already depends on it.
+        warnings.warn('Creating a padded array with longer length than '
```

Comment:
I think this `max_length > max(lengths)` should be the normal situation. Don't understand why warning here?

Reply:
Good catch. This logic is backwards. I'll add a test.